### PR TITLE
Support per-host tokens

### PR DESCRIPTION
**Pull request by colinking**
This is useful for our internal development, where we test the CLI against a local/staging/production host. Previously tokens were global, but now they are per-host so there will be a token persisted for each host you've used.

### CODE REVIEW

File: pkg/cmd/auth/logout/logout.go [hunk 1]
```diff
@@ -23,12 +23,14 @@ func New(c *cli.Config) *cobra.Command {
...
 	cfg, err := conf.ReadDefault()
-	if !errors.Is(err, conf.ErrMissing) {
-		cfg.Token = ""
+	if err == nil || !errors.Is(err, conf.ErrMissing) {
+		delete(cfg.Tokens, c.Client.Host)
 
 		if err := conf.WriteDefault(cfg); err != nil {
 			return err
 		}
+	} else if err != nil {
```
Comment (joshma) [inline pkg/cmd/auth/logout/logout.go:32]: so this only returns the err if it IS `conf.ErrMissing`? why this change?

Comment (colinking) [reply pkg/cmd/auth/logout/logout.go]: Ah yeah, the logic here for ErrMissing was wrong. Thanks for catching! Made the check simpler.
